README.md

```diff
@@ -1396,7 +1396,56 @@ string isn't available in the _data_, the message is _not_ published.
 
 ### `mysql`
 
-The MySQL plugin is one of the most complicated to set up. It requires the following configuration:
+The MySQL plugin will attempt to add a row for every message received on a given topic, automatically filling in
+ columns.
+
+For instance, given a table created with `CREATE TABLE names (id INTEGER, name VARCHAR(25));` then
+the message '{ "name" : "Jane Jolie", "id" : 90, "number" : 17 }' on topic 'my/2' will be added to the table like this:
+
+```mysql
++------+------------+
+| id   | name       |
++------+------------+
+|   90 | Jane Jolie |
++------+------------+
+```
+
+The values for the 'id' and 'name' columns are assumed to be filled by the values of the JSON nodes with the same name.
+
+If you added columns 'topic', 'payload' and '_dtiso' to the databse, then that same message will add this row:
+
+```mysql
++------+------------+-----------------------------------------------------+-----------------------------+-------+
+| id   | name       | payload                                             | _dtiso                      | topic |
++------+------------+-----------------------------------------------------+-----------------------------+-------+
+|   90 | Jane Jolie | { "name" : "Jane Jolie", "id" : 90, "number" : 17 } | 2018-09-17T20:20:31.889002Z | my/2  |
++------+------------+-----------------------------------------------------+-----------------------------+-------+
+```
+Here, the plugin pulled values for the new columns from standard mqttwarn meta-data.
+
+When a message is received, the  plugin will attempt to populate the following column names:
+- root-level JSON nodes in the message
+  - e.g. 'name' and 'id' above
+- ['transformation data' fields](https://github.com/jpmens/mqttwarn#outbound-messages) names
+  - e.g. 'topic', 'payload' and '_dtiso' as above
+  - note that these all must be VARCHAR columns; timestamp columns are [not yet supported](https://github.com/jpmens/mqttwarn/issues/334#issuecomment-422141808)
+- the 'fallback' column, as noted below
+  - #TODO: how is this different from 'payload'?
```

Comment:
@jpmens ?

Reply:
the fallback column gets all the JSON elements which have **not** been otherwise stored. e.g. if `payload` contains `{ name: Jane, number: 42 }` (think JSON please), and we have a column `name` in the DB, then fallback will get `{number: 42}` whereas `payload` would have the full original payload.

Reply:
I just tried it, and it _seems_ that 'payload' and 'full' are the same.  I added the 'full' column and posted a new message:
```
|   90 | Jane Jolie | { "name" : "Jane Jolie", "id" : 90, "number" : 17 } | NULL                                                | 2018-09-17T20:20:31.889002Z | db    |
|   91 | Jane Jolie | { "name" : "Jane Jolie", "id" : 91, "number" : 17 } | { "name" : "Jane Jolie", "id" : 91, "number" : 17 } | 2018-09-18T20:44:51.662836Z | db    |
```

Looking at the code, I see where the fallback column is set: https://github.com/jpmens/mqttwarn/blob/master/services/mysql.py#L67

But there's no point where fields are removed.  We only know which fields should be removed _after_ `add_row` is called, which is too late.